lib/fastlane/actions/add_icon_overlay.rb

```diff
@@ -0,0 +1,181 @@
+            c.pointsize "#{text_size}" # set font size
+          end
+
+          result.write icon
+        end
+      end
+
+      def self.addImageOverlay(appiconset, params)
+        overlay_image = MiniMagick::Image.new(params[:overlay_image_path])
+
+        Dir.glob(File.join(appiconset, '*.png')) do |icon|
+          original_icon = MiniMagick::Image.new(icon)
+
+          result = original_icon.composite(overlay_image) do |c|
+            c.compose "Over"
+            c.resize "#{original_icon.width}x#{original_icon.height}"
+            c.quality 100
+          end
+          result.write icon
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Overlays an image or text over all icons."
+      end
+
+      def self.details
+        "Overlays an image or text over all icons, using ImageMagick and replacing the old icons.
+        Text overlay based off @merowing's script http://merowing.info/2013/03/overlaying-application-version-on-top-of-your-icon/
+        Banner overlay based off of @squarefrog's comment https://github.com/KrauseFx/fastlane/issues/481#issuecomment-147368555"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :appiconset_path,
+                                       env_name: "FL_ADD_ICON_OVERLAY_APPICONSET_PATH",
+                                       description: "Path for .appiconset containing the icons",
+                                       is_string: true,
+                                       default_value: '**/AppIcon.appiconset'
+                                       ),
+          FastlaneCore::ConfigItem.new(key: :overlay_image_path,
+                                       env_name: "FL_ADD_ICON_OVERLAY_OVERLAY_IMAGE_PATH",
+                                       description: "Path for the image that will be overlaid on the icons",
+                                       is_string: true,
+                                       optional: true,
+                                       verify_block: Proc.new do |value|
+                                       raise "No overlay image path for AddIconOverlayAction given, pass using `overlay_image_path: 'image_path.png'`".red unless (value and not value.empty?)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :overlay_type,
+                                       env_name: "FL_ADD_ICON_OVERLAY_OVERLAY_TYPE",
+                                       description: "Flag specifying if an image, text or banner will be overlayed over the icon",
+                                       is_string: true,
+                                       default_value: 'text',
+                                       verify_block: Proc.new do |value|
```

Comment:
Already fixed with e0e2bb3f35cfd00ad9401eea4a7e4b72dffcb87a but `HoundCI` failed to pick the change